R/execution.r

```diff
@@ -120,6 +120,28 @@ quit = function(save = 'default', status = 0, runLast = TRUE) {
     payload <<- c(.self$payload, list(list(source = 'ask_exit', keepkernel = FALSE)))
 },
 
+# noninteractive
+readline = function(prompt='') {
+    log_debug("entering custom readline")
```

Comment:
code style (here and below); please use:

* spaces around opearators and default argument `=`
* single quotes, except in strings containing `'` or method docstrings
* `<-` for assignment
* no useless `return`, i.e. only use `return` for early exit, never in terminal position